### PR TITLE
Landing page

### LINKED ISSUE #35: The system can support a landing page to collect email addresses of interested users pre-launch

### PR DESCRIPTION
**Pull request by littleforest**
closes #35 

## Description of changes
* Set up RSpec for testing suite
* Add user model
* Allow Admins to update users
* Create landing page that allows visitors to enter email address and name

## Notes
This is pretty basic at the moment -- it just accepts any email address and if it doesn't exist, it will create a new user. Otherwise it just returns gracefully with no errors.  Let me know if you want more functionality, or we can create some follow-up tickets as needed.  My CSS is a little clunky, just heads up.

### CODE REVIEW

**Comment by Pcushing** [comment]
@littleforest one request... anything that's front-end, can you include a snapshot(s) in the PR? 

That often allows me to give quick feedback even if I'm not in a position to pull the whole branch down to run the code in the branch.

**Comment by littleforest** [comment]
Sounds good!  Here is the screenshot for the landing page:

Screen Shot 2019-09-16 at 4 00 23 PM

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        post "/welcome", params: params
+        expect(response).to have_http_status(:success)
+        expect(response.body).to match "enter your email address"
+      end
+    end
+
+    context "with email already in use" do
+      let!(:user) { create(:user, email: "foo@example.com", name: "Foo Bar") }
+
+      it "does not create new user record" do
```
Comment (Pcushing) [inline spec/requests/users_spec.rb]: One side note - I love the testing. We haven't talked about _what_ we're going to test so for now just go with what you're doing. It's instructive. 

This is a simple example where it doesn't really matter, but I see this request spec and think it does the same as the model spec for email validation. The page rendering is unique, but the `does not create new user record` should be covered if we trust the other spec, right? Or does this cover something that doesn't?

As this gets more complicated, we might target certain types of tests and not others is the only point. What do you think? Let's plan to discuss live in our next update, and we'll go from there.

**Comment by Pcushing** [comment]
Straightforward to me. Merge away!

File: spec/requests/users_spec.rb (hunk 1 above)
Comment (littleforest) [reply spec/requests/users_spec.rb]: That's true -- the actual check for whether the user count increases we could probably skip, but I think the check that the name doesn't get updated is important.  I could have coded it that entering the same email twice but with a different name would have overwritten the original name.

Sure, we can discuss tests next time.  I generally try to have pretty good test coverage.

**Pull request merged**